### PR TITLE
WIP: Use Time APIs for date in the Request.java and Warning value.

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/ResponseProtocolCompliance.java [hunk 1]
```diff
@@ -101,8 +100,8 @@ private void warningsWithNonMatchingWarnDatesAreRemoved(
         boolean modified = false;
         for(final Header h : warningHeaders) {
             for(final WarningValue wv : WarningValue.getWarningValues(h)) {
-                final Date warnDate = wv.getWarnDate();
-                if (warnDate == null || warnDate.equals(responseDate)) {
+                final Instant warnInstant = wv.getWarnInstant();
+                if (warnInstant == null || warnInstant.equals(responseDate)) {
                     newWarningHeaders.add(new BasicHeader(HeaderConstants.WARNING,wv.toString()));
```
Comment (michael-o) [inline httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/ResponseProtocolCompliance.java:105]: A bit confusing, `getWarnDate()` has been renamed, but `responseDate` not?

Comment (arturobernalg) [reply httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/ResponseProtocolCompliance.java:105]: Hi @michael-o 
surely you have. reason, but the idea is to do it in stages.
TY

Comment (michael-o) [reply httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/ResponseProtocolCompliance.java:105]: OK

Comment (arturobernalg) [reply httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/ResponseProtocolCompliance.java:105]: Thank you @michael-o 
I just want avoid mistake and mess the code.   My idea is make the apis change progressively. 
TY

**Review by michael-o — APPROVED** [review]

File: httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/WarningValue.java [hunk 2]
```diff
@@ -343,8 +346,18 @@ private void parseError() {
      * {@code null} if a warning date was not supplied in the
      * header.
      * @return {@link Date}
+     * @deprecated Use {@link #getWarnInstant()}
+     */
+    @Deprecated
```
Comment (ok2c) [inline httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/WarningValue.java:351]: @arturobernalg  This is all unnecessary. The class is not public.

**Review by ok2c — CHANGES_REQUESTED** [review]

**Comment by ok2c** [comment]
@arturobernalg 

1. Please do not feed your changes piecemeal. Work on a branch and raise a PR once _everything_ is ready. If you need help or another pair of eyeballs, ask people to look at the code in your branch
2. Only public classes need to remain backward compatible. Private and package private classes can be changed without deprecation and `@since` metadata.
3. If you are concerned about breaking existing functionality, create a test case with old APIs, make sure it works, then port it to the new APIs.

**Comment by arturobernalg** [comment]
> @arturobernalg
> 
> 1. Please do not feed your changes piecemeal. Work on a branch and raise a PR once _everything_ is ready. If you need help or another pair of eyeballs, ask people to look at the code in your branch
> 2. Only public classes need to remain backward compatible. Private and package private classes can be changed without deprecation and `@since` metadata.
> 3. If you are concerned about breaking existing functionality, create a test case with old APIs, make sure it works, then port it to the new APIs.

HI @ok2c 
Fair enough. I change the PR to draft until I change all the api.
TY

**Comment by arturobernalg** [comment]
Close this PR in favor of [HTTPCLIENT-2189](https://issues.apache.org/jira/browse/HTTPCLIENT-2189) that change all code the remains HttpClient API.

**Comment by arturobernalg** [comment]
Close this PR in favor the [HTTPCLIENT-2189](https://github.com/apache/httpcomponents-client/pull/330)

**Pull request closed without merging**